proto/cosmos/streaming/v1/grpc.proto

```diff
@@ -0,0 +1,80 @@
+syntax = "proto3";
+
+package cosmos.streaming.v1;
+
+import "gogoproto/gogo.proto";
+import "google/protobuf/timestamp.proto";
+
+option go_package = "cosmossdk.io/server/v2/streaming";
+
+// ListenDeliverBlockRequest is the request type for the ListenDeliverBlock RPC method
+message ListenDeliverBlockRequest {
+  repeated Transaction      txs                  = 1;
+  int64                     height               = 2;
+  repeated Event            events               = 3;
+  bytes                     hash                 = 4;
+  google.protobuf.Timestamp time                 = 5 [(gogoproto.stdtime) = true];
+  bytes                     next_validators_hash = 6;
+  bytes                     proposer_address     = 7;
+}
+
+// ListenDeliverBlockResponse is the response type for the ListenDeliverBlock RPC method
+message ListenDeliverBlockResponse {}
+
+// ListenCommitRequest is the request type for the ListenCommit RPC method
+message ListenStateChangesRequest {
+  // explicitly pass in block height as ResponseCommit does not contain this
+  // info
+  int64                block_height = 1;
+  repeated StoreKVPair change_set   = 2;
+  bytes                app_hash     = 3;
+}
+
+// ListenStateChangesResponse is the response type for the ListenCommit RPC method
+message ListenStateChangesResponse {}
+
+// ListenerService is the service for the ABCIListener interface
+service ListenerService {
+  // ListenDeliverBlock is the corresponding endpoint for ABCIListener.ListenDeliverBlock
+  rpc ListenDeliverBlock(ListenDeliverBlockRequest) returns (ListenDeliverBlockResponse);
+  // ListenCommit is the corresponding endpoint for ABCIListener.ListenCommit
+  rpc ListenStateChanges(ListenStateChangesRequest) returns (ListenStateChangesResponse);
+}
+
+// StoreKVPair is a single key-value pair, associated with a store.
+message StoreKVPair {
+  string store_key = 1; // the store key for the KVStore this pair originates from
+  bool delete      = 2; // true indicates a delete operation, false indicates a set operation
```

Comment:
we can delete this -- nil value should indicate a delete

Reply:
i removed a few more fields in favor on later introducing a listenBlock endpoint in which the entire block is streamed out.